### DISCUSSION TITLE
Run shell command with pipeline

### DISCUSSION THREAD

**Discussion by aa1ss2** [helix-editor/helix#7780] [General]
I want to pipeline each selection to the shell command and display the returned results (like :sh retruned),What should I do?
Used :pipe will changed selection,
Cannot write to pipeline with :sh command,
What is the register to obtain the value of the selection area?

**Comment by pascalkuthe** [comment]
`:pipe-to`

**Comment by aa1ss2** [comment]
thx for answering
:pipe-to do not display results
for example ,used wc to count selection words,

I can only use :pipe wc --word ,and press u ,is there a better way?

**Comment by docteurklein** [comment]
I agree that the current behavior seems weird to me (helix newcomer) coming from vim where selections, current-filename (%) and such are first-class citizens in **any** command prompt.

Looks like we could relax the constraints here: https://github.com/helix-editor/helix/blob/77fe8f214b15b8297a9f2f8ed929994559604d6c/helix-term/src/commands.rs#L5473-L5476

Specifically, allow any shell command to do those things to the command output:
   - insert (currently `insert-output`)
   - append (currently `append-output`)
   - replace (currently `pipe`)
   - ignore (currently `pipe-to` and `run-shell-command`, although the latter has the side-effect of showing a modal)

But more importantly, all of them **should be able to receive the selections as stdin**, which is currently not supported due to the aforementioned constraint.

If we were to relax it, the variety of commands could be maybe reduced to a simple modifier to describe the operation (insert, append, …) followed by `:pipe`, following (to me at least) the helix/kakoune philosophy, no?

**Comment by djpowers** [comment]
I agree that this would be a handy feature. It looks like #3134 could define a selection substitution as well to cover this use case.

**Comment by Euphrasiologist** [comment]
Is there any update on this? Or still `:pipe` over writing selected text?

**Comment by weiying-chen** [comment]
I also want to be able to select text in Helix (like with `:pipe`) and return a result (like with `:sh`). Right now, the closest thing I can do, is to do `o = ["yank_to_clipboard", ":sh ~/bash/script.sh"]`.

**Comment by dkonsoul** [comment]
Are there any updates to this? Maybe the pipe-to (or maybe a new pipe-to-register), instead of ignoring the output fully, it could write it on a register, so we can paste it if needed.

**Comment by lihs-learning** [comment]
Migrate from vim. I miss somethings like `w !wc` and `'<,'>! wc`, see `:help write_c`.
